src/v3/plugins/github/createGraph.js

```diff
@@ -1,161 +1,95 @@
 // @flow
 
 import {Graph} from "../../core/graph";
-import type {
-  GithubResponseJSON,
-  RepositoryJSON,
-  ReviewJSON,
-  PullJSON,
-  IssueJSON,
-  CommentJSON,
-  ReviewCommentJSON,
-  NullableAuthorJSON,
-} from "./graphql";
-
-import type {
-  RepoAddress,
-  IssueAddress,
-  PullAddress,
-  ReviewAddress,
-  UserlikeAddress,
-  StructuredAddress,
-  AuthorableAddress,
-  ChildAddress,
-  ParentAddress,
-} from "./nodes";
-import {toRaw} from "./nodes";
-
+import * as GitNode from "../git/nodes";
+import * as N from "./nodes";
+import * as R from "./relationalView";
 import {createEdge} from "./edges";
 
-import {COMMIT_TYPE, toRaw as gitToRaw} from "../git/nodes";
-
-import {
-  reviewUrlToId,
-  issueCommentUrlToId,
-  pullCommentUrlToId,
-  reviewCommentUrlToId,
-} from "./urlIdParse";
-
-export function createGraph(data: GithubResponseJSON): Graph {
-  const creator = new GraphCreator();
-  creator.addData(data);
+export function createGraph(view: R.RelationalView): Graph {
+  const creator = new GraphCreator(view);
   return creator.graph;
 }
 
 class GraphCreator {
   graph: Graph;
+  view: R.RelationalView;
 
-  constructor() {
+  constructor(view: R.RelationalView) {
     this.graph = new Graph();
+    this.view = view;
+    for (const r of view.repos()) {
```

Comment:
Is there a reason to do this in the constructor? All else equal,
I prefer the previous pattern of an `addData` method that’s called by
the `createGraph` function. It’s nice when (a) constructors don’t do any
work, and (b) constructors don’t call instance methods. The former is
mostly aesthetic (in JavaScript, at least), but the latter can be
important: instance methods may assume that class invariants hold, but
these class invariants may not be established before the constructor has
finished.

Reply:
I changed it to suit your preference.